### PR TITLE
Plane: add option to entirely disable short failsafe

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduPlane/system.cpp [hunk 1]
```diff
@@ -512,15 +512,17 @@ void Plane::check_short_failsafe()
 {
     // only act on changes
     // -------------------
-    if(failsafe.state == FAILSAFE_NONE && flight_stage != AP_Vehicle::FixedWing::FLIGHT_LAND) {
+    if (g.short_fs_action != SHORT_FS_ACTION_DISABLED &&
+       failsafe.state == FAILSAFE_NONE &&
+       flight_stage != AP_Vehicle::FixedWing::FLIGHT_LAND) {
```
Comment (magicrub) [inline ArduPlane/system.cpp]: don't forget stage land abort

**Comment by peterbarker** [comment]
> don't forget stage land abort

Wasn't even on my radar.  Pre-existing issue, get your own PR :-P

**Comment by peterbarker** [comment]
@magicrub That's an unrelated change with implications I can't even pretend to understand at the moment.

Yes, it's a good change.  And I might even attack it after doing this PR.

But does it really need to be done as a prerequisite for this one to go in?

I'd point out that at a minimum the long failsafe probably also requires the same modifications so factoring out a `fs_inhibittedfor_flightmode(AP_Vehicle::FixedWing::FlightStage fs)` method and and and...

File: ArduPlane/events.cpp [hunk 2]
```diff
@@ -61,7 +61,7 @@ void Plane::failsafe_short_on_event(enum failsafe_state fstype, mode_reason_t re
 void Plane::failsafe_long_on_event(enum failsafe_state fstype, mode_reason_t reason)
 {
     // This is how to handle a long loss of control signal failsafe.
-    gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe. Long event on, ");
+    gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe. Long event on: type=%u/reason=%u", fstype, reason);
```
Comment (WickedShell) [inline ArduPlane/events.cpp]: This line is partially conveying the same information as events.cpp:118, one should be removed for the other.

File: ArduPlane/events.cpp [hunk 3]
```diff
@@ -5,7 +5,7 @@ void Plane::failsafe_short_on_event(enum failsafe_state fstype, mode_reason_t re
     // This is how to handle a short loss of control signal failsafe.
     failsafe.state = fstype;
     failsafe.ch3_timer_ms = millis();
-    gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe. Short event on, ");
+    gcs().send_text(MAV_SEVERITY_WARNING, "Failsafe. Short event on: type=%u/reason=%u", fstype, reason);
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply ArduPlane/events.cpp]: I actually started making these strings but decided it was a lot of code size for little reward.  I did think it worthwhile to provide more debug, however.

You think this is a step backwards, so I'll remove the patch.

File: ArduPlane/events.cpp (hunk 2 above)
Comment (peterbarker) [reply ArduPlane/events.cpp]: Removed the other one.

File: ArduPlane/system.cpp [hunk 4]
```diff
@@ -480,8 +480,13 @@ void Plane::check_long_failsafe()
     // only act on changes
     // -------------------
     if (failsafe.state != FAILSAFE_LONG && failsafe.state != FAILSAFE_GCS && flight_stage != AP_Vehicle::FixedWing::FLIGHT_LAND) {
-        if (failsafe.state == FAILSAFE_SHORT &&
-                   (tnow - failsafe.ch3_timer_ms) > g.long_fs_timeout*1000) {
+        uint32_t radio_timeout_ms = failsafe.last_valid_rc_ms;
+        if (g.short_fs_action != SHORT_FS_ACTION_DISABLED) {
+            // time is relative to when short failsafe enabled
+            radio_timeout_ms = failsafe.ch3_timer_ms;
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply ArduPlane/system.cpp]: I am working on the assumption that the short failsafe will be scheduled ahead of the long failsafe.  I believe this is a property of the current scheduler and declaration order.  It's an interesting landmine to add, however.

I added a check that we're actually in the short failsafe instead, which is much more sensible, I think!

**Pull request merged**